lib/test_prof/event_prof/profiler.rb

```diff
@@ -40,6 +41,7 @@ def track(time)
 
         @example_time += time
         @example_count += 1
+        @total_test_time += example_time
```

Comment:
The `time` passed to `track` method is the event time.

We can try to extend `track` method to accept two arguments: `event_time` (which we track right now) and `total_time` (the total time example ran).

Reply:
I'm not sure how we can capture the total_time. Can you elaborate a little but more?

Reply:
> how we can capture the total_time.

Smth like this:

```ruby
def initialise
  # ...
  @total_run_time = 0.0
end

def group_finished(id)
  group_run_time = TestProf.now - @group_ts

  @total_run_time += group_run_time
  data = { id: id, time: @time, run_time: group_run_time, count: @count, examples: @total_examples }
  @groups << data unless data[rank_by].zero?

  @current_group = nil
end

def example_finished(id)
  @total_examples += 1
  return unless per_example?
        
  data = { id: id, time: @example_time, run_time: TestProf.now - @example_ts, count: @example_count }
  @examples << data unless data[rank_by].zero?
  @current_example = nil
end

def reset_group!
  @time = 0.0
  @count = 0
  @total_examples = 0
  @group_ts = TestProf.now
end

def reset_example!
  @example_count = 0
  @example_time = 0.0
  @example_ts = TestProf.now
end
```

Reply:
I understande the general logic but wouldn't `@time` be the same as the `group_run_time`?

Reply:
> wouldn't @time be the same as the group_run_time

Unlikely.

`@time` is the time spending _within an event_, e.g. time spent for database queries.

`run_time` is the whole example/group time, so it is always greater than `@time`, 'cause it contains it

Reply:
Got it

Reply:
> > how we can capture the total_time.
> 
> Smth like this:
> 
> ```ruby
> def initialise
>   # ...
>   @total_run_time = 0.0
> end
> 
> def group_finished(id)
>   group_run_time = TestProf.now - @group_ts
> 
>   @total_run_time += group_run_time
>   data = { id: id, time: @time, run_time: group_run_time, count: @count, examples: @total_examples }
>   @groups << data unless data[rank_by].zero?
> 
>   @current_group = nil
> end
> 
> def example_finished(id)
>   @total_examples += 1
>   return unless per_example?
>         
>   data = { id: id, time: @example_time, run_time: TestProf.now - @example_ts, count: @example_count }
>   @examples << data unless data[rank_by].zero?
>   @current_example = nil
> end
> 
> def reset_group!
>   @time = 0.0
>   @count = 0
>   @total_examples = 0
>   @group_ts = TestProf.now
> end
> 
> def reset_example!
>   @example_count = 0
>   @example_time = 0.0
>   @example_ts = TestProf.now
> end
> ```

I think I'm getting this wrong. On the tests I'm always getting small times because there's no real events. Who can I test this on a real situations

Reply:
> On the tests I'm always getting small times because there's no real events

Yeah, tests use non-real timings. For real events that shouldn't be the case though.

One way is to add `sleep` statements to https://github.com/palkan/test-prof/blob/master/spec/integrations/fixtures/rspec/event_prof_fixture.rb#L12 (and other fixtures) to make events timings more realistic (and likely change all the value to be less than a second (i.e. 240 -> 0.24)).

Reply:
> The change that you spoke before about the track method implies that the event already brings the total_time together with the time

With the above approach we don't need to modify the `track` method

Reply:
I'll get to it when I have the time, which doesn't come often. Sorry for the delay!